fontbe/src/features/marks.rs

```diff
@@ -462,6 +567,80 @@ impl<'a> MarkLookupBuilder<'a> {
         let name = self.glyph_order.glyph_name(gid.into()).cloned().unwrap();
         Error::AnchorDeltaError(name, err)
     }
+
+    fn scripts_using_abvm(&self) -> HashSet<UnicodeShortName> {
+        let fea_scripts = super::get_script_language_systems(&self.fea_first_pass.ast)
+            .into_keys()
+            .collect::<HashSet<_>>();
+        INDIC_SCRIPTS
+            .iter()
+            .chain(USE_SCRIPTS)
+            .chain(std::iter::once(&"Khmr"))
+            .filter_map(|s| UnicodeShortName::from_str(s).ok())
+            .filter(|script| fea_scripts.is_empty() || fea_scripts.contains(script))
+            .collect()
+    }
+
+    //https://github.com/googlefonts/ufo2ft/blob/5a606b7884bb6da/Lib/ufo2ft/featureWriters/markFeatureWriter.py#L1119
+    //
+    // returns two sets: glyphs used in abvm/blwm, and glyphs used in mark
+    // (in theory these could intersect?)
+    fn split_mark_and_abvm_blwm_glyphs(
+        &self,
+    ) -> Result<(HashSet<GlyphId16>, HashSet<GlyphId16>), Error> {
+        let scripts_using_abvm = self.scripts_using_abvm();
+
+        let unicode_is_abvm = |uv: u32| -> bool {
+            super::properties::scripts_for_codepoint(uv)
+                .any(|script| scripts_using_abvm.contains(&script))
+        };
+
+        // note that it's possible for a glyph to pass both these tests!
+        let unicode_is_non_abvm = |uv: u32| -> bool {
+            super::properties::scripts_for_codepoint(uv)
+                .any(|script| !scripts_using_abvm.contains(&script))
+        };
+
+        if scripts_using_abvm.is_empty() || !self.char_map.keys().copied().any(unicode_is_abvm) {
+            // no abvm scripts: everything is a mark
+            return Ok((
+                Default::default(),
+                self.glyph_order.iter().map(|(gid, _)| gid).collect(),
+            ));
+        }
+
+        let gsub = self.fea_first_pass.gsub();
+        let abvm_glyphs = super::properties::glyphs_matching_predicate(
+            &self.char_map,
+            unicode_is_abvm,
+            gsub.as_ref(),
+        )?;
+
+        let mut non_abvm_glyphs = super::properties::glyphs_matching_predicate(
+            &self.char_map,
+            unicode_is_non_abvm,
+            gsub.as_ref(),
+        )?;
+        // https://github.com/googlefonts/ufo2ft/blob/5a606b7884bb6da/Lib/ufo2ft/featureWriters/markFeatureWriter.py#L1156
+        // I do not understand why we have all this fancy logic when it feels like
+        // 'not abvm' is implicit?
```

Comment:
because https://github.com/googlefonts/ufo2ft/blob/16ed156bd6a8b9bc035d0aa8045a1271ef79a52e/Lib/ufo2ft/featureWriters/markFeatureWriter.py#L1148-L1150